src/sentry/workflow_engine/endpoints/validators/base/action.py

```diff
@@ -0,0 +1,60 @@
+from typing import Any, Generic, TypeVar
+
+from django.forms import ValidationError
+from jsonschema import ValidationError as JsonValidationError
+from jsonschema import validate
+from rest_framework import serializers
+
+from sentry.api.serializers.rest_framework import CamelSnakeModelSerializer
+from sentry.db.models import Model
+from sentry.workflow_engine.models import Action
+from sentry.workflow_engine.registry import action_handler_registry
+from sentry.workflow_engine.types import ActionHandler
+
+# from sentry.utils.registry import NoRegistrationExistsError
+
+T = TypeVar("T", bound=Model)
+ActionData = dict[str, Any]
+ActionConfig = dict[str, Any]
+
+
+def validate_json_schema(value, schema):
+    try:
+        validate(value, schema)
+    except JsonValidationError as e:
+        raise ValidationError(str(e))
+
+    return value
+
+
+class BaseActionValidator(CamelSnakeModelSerializer[T], Generic[T]):
+    data: Any = serializers.JSONField()
+    config: Any = serializers.JSONField()
+
+    class Meta:
+        model = T
+        fields = ["config", "data", "integration_id", "type"]
```

Comment:
do we need `fields` if this a incoming validator, not an outgoing serializer? iiuc `fields` is used for specifying the fields to serialize if returning something in a `Response`

Reply:
Yes - w/o fields it doesn't run the `validate_<field>` methods, found that one out the hard way early on. haha

Reply:
dang that's crazy